### PR TITLE
Change signature for read_region

### PR DESCRIPTION
**Pull request by jonasteuwen**
This PR prepares for the upcoming dlup 0.4.

### CODE REVIEW

File: ahcore/readers.py [hunk 1]
```diff
@@ -325,6 +263,31 @@ def read_region_raw(self, location: tuple[int, int], size: tuple[int, int]) -> G
 
         return stitched_image
 
+    def read_region(self, location: tuple[int, int], level: int, size: tuple[int, int]) -> GenericNumberArray:
```
Comment (VanessaBotha) [inline ahcore/readers.py:266]: Do I understand this correctly? 
scaling=1 is the same as level=0 and has it now been decided to not support scaling other than 1 (therefore all other code in the old read_region function is no longer needed)?

Comment (VanessaBotha) [reply ahcore/readers.py:266]: Are there no places in the ahcore codebase where the (old) .read_region function is called? 
In the new code, the scaling argument has been changed to level. 
Assuming that scaling=1 is the same as level=0, I would expect this should be changed in places where read_region is called to preserve the same functionality. I don't see this in the changes 
(I only see replacement of .read_region_raw --> .read_region)

**Review by VanessaBotha — COMMENTED** [review]
It looks good to me, but had some questions

File: ahcore/readers.py (hunk 1 above)
Comment (jonasteuwen) [reply ahcore/readers.py:266]: Hi @VanessaBotha,

I think I’ve replaced the calls with the new signature. The new signature is compatible with the openslide read_region. We have a SlideImage class that wraps it to a function with a scaling parameter. So eventually it is the idea to replace the class with a Backend so we can read at an arbitrary resolution (scaling). That will be dlup, and is now a PR

Comment (VanessaBotha) [reply ahcore/readers.py:266]: Oke clear! 
Regarding the signature change: what I meant is that I expected changes like this to be needed in places where the old .read_region function was called in ahcore:
.read_region(<location>, 1, <size>) —> .read_region(<location>, 0, <size>)
Unless the old .read_region was never used in ahcore

**Pull request closed without merging**